components/widgets/badges/badge.tsx

```diff
@@ -6,7 +6,13 @@ import PropTypes from 'prop-types';
 
 import './badge.scss';
 
-export default class Badge extends React.Component {
+type Props = {
+    show: boolean,
+    children: React.ReactNode,
+    className: string,
+};
+
+export default class Badge extends React.Component<Props> {
     static propTypes = {
```

Comment:
propTypes are not needed when using TSX. it's handled automatically and redundant

Reply:
The propTypes are needed because we still have javascript code calling to this, the propTypes are not needed when everything is typescript because you can detect the same kind of errors at compile time. So I think we should maintain the propTypes here, or search an "automatic way" to transform them into propTypes.

Reply:
In that case, they should be properly typed as well: https://dev.to/busypeoples/notes-on-typescript-inferring-react-proptypes-1g88